Panic in cloudflare_access_group with empty email list

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.0.3
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v2.24.0

### Affected resource(s)

cloudflare_access_group

### Terraform configuration files

```hcl
resource "cloudflare_access_group" "group" {
  zone_id = "..."
  name    = "group"

  include {
    email = []
  }
}
```


### Debug output

none

### Panic output

╷
│ Error: Plugin did not respond
│ 
│   with cloudflare_access_group.group["test"],
│   on cloudflare-access.tf line 41, in resource "cloudflare_access_group" "group":
│   41: resource "cloudflare_access_group" "group" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-cloudflare_v2.24.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 13 [running]:
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.appendConditionalAccessGroupFields(0xc0004325a0, 0x24, 0x0, 0x0, 0xc0006d6410, 0xc, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_access_group.go:407 +0x31b
github.com/cloudflare/terraform-provider-cloudflare/cloudflare.resourceCloudflareAccessGroupUpdate(0xc00020d1f0, 0x11492c0, 0xc00044a240, 0x24, 0x195aa40)
	github.com/cloudflare/terraform-provider-cloudflare/cloudflare/resource_cloudflare_access_group.go:321 +0x178
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00029f220, 0xc0004587d0, 0xc0005f0400, 0x11492c0, 0xc00044a240, 0x1003b01, 0xc0004178d0, 0xc00072aae0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:326 +0x273
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00060e300, 0xc000417a38, 0xc0004587d0, 0xc0005f0400, 0xc0006b5b88, 0xc000685170, 0x1005cc0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x99
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000e290, 0x1344b30, 0xc0000d5920, 0xc00020c620, 0xc00000e290, 0xc0000d5920, 0xc00083aba0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x8a5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x110eb40, 0xc00000e290, 0x1344b30, 0xc0000d5920, 0xc0005fe660, 0x0, 0x1344b30, 0xc0000d5920, 0xc00004a700, 0x349)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0006bc1c0, 0x134e2d8, 0xc000103e00, 0xc000832100, 0xc000549470, 0x1918cc0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0006bc1c0, 0x134e2d8, 0xc000103e00, 0xc000832100, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0006d6190, 0xc0006bc1c0, 0x134e2d8, 0xc000103e00, 0xc000832100)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-cloudflare_v2.24.0 plugin crashed!

### Expected output

no crash

### Actual output

crash

### Steps to reproduce

create a group with an empty email list

### Additional factoids

_No response_

### References

_No response_

--- a/.changelog/1237.txt
+++ b/.changelog/1237.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_access_policy: handle empty `nil` values for building policies
+```

--- a/cloudflare/resource_cloudflare_access_group.go
+++ b/cloudflare/resource_cloudflare_access_group.go
@@ -394,17 +394,23 @@ func resourceCloudflareAccessGroupImport(d *schema.ResourceData, meta interface{
 func appendConditionalAccessGroupFields(group cloudflare.AccessGroup, d *schema.ResourceData) cloudflare.AccessGroup {
 	exclude := d.Get("exclude").([]interface{})
 	for _, value := range exclude {
-		group.Exclude = BuildAccessGroupCondition(value.(map[string]interface{}))
+		if value != nil {
+			group.Exclude = BuildAccessGroupCondition(value.(map[string]interface{}))
+		}
 	}
 
 	require := d.Get("require").([]interface{})
 	for _, value := range require {
-		group.Require = BuildAccessGroupCondition(value.(map[string]interface{}))
+		if value != nil {
+			group.Require = BuildAccessGroupCondition(value.(map[string]interface{}))
+		}
 	}
 
 	include := d.Get("include").([]interface{})
 	for _, value := range include {
-		group.Include = BuildAccessGroupCondition(value.(map[string]interface{}))
+		if value != nil {
+			group.Include = BuildAccessGroupCondition(value.(map[string]interface{}))
+		}
 	}
 
 	return group

